be/src/storage/lake/delta_writer.cpp

```diff
@@ -282,19 +290,30 @@ inline Status DeltaWriterImpl::flush_async() {
         if (_miss_auto_increment_column && _mem_table->get_result_chunk() != nullptr) {
             RETURN_IF_ERROR(_fill_auto_increment_id(*_mem_table->get_result_chunk()));
         }
-        st = _flush_token->submit(std::move(_mem_table));
-        _mem_table.reset(nullptr);
-        if (_immutable_tablet_size > 0) {
-            _tablet_manager->set_in_writing_data_size(_tablet_id, _txn_id, _tablet_writer->data_size());
-            ASSIGN_OR_RETURN(auto tablet, _tablet_manager->get_tablet(_tablet_id));
-            VLOG(1) << "flush memtable, tablet=" << _tablet_id << ", txn=" << _txn_id
-                    << " _immutable_tablet_size=" << _immutable_tablet_size
-                    << ", writer_data_size=" << _tablet_writer->data_size()
-                    << ", tablet_data_size=" << tablet.data_size() + _tablet_manager->in_writing_data_size(_tablet_id);
-            if (tablet.data_size() + _tablet_manager->in_writing_data_size(_tablet_id) > _immutable_tablet_size) {
-                _is_immutable.store(true, std::memory_order_relaxed);
+        st = _flush_token->submit(std::move(_mem_table), false, [this](std::unique_ptr<SegmentPB> seg, bool eos) {
+            if (seg) {
+                _tablet_manager->add_in_writing_data_size(_tablet_id, _txn_id, seg->data_size());
             }
-        }
+            if (_immutable_tablet_size > 0) {
```

Comment:
1. What's the difference between `_immutable_tablet_size` and `min_immutable_tablet_size`?
2. What's their usage?

Reply:
make it same